connectors/cache/fallback_test.go

```diff
@@ -115,41 +104,44 @@ func TestUpsertCases(t *testing.T) {
 
 		mockOrigin.EXPECT().Upsert(context.TODO(), testEi, tc.originUpsert.values).Return(nil)
 		if tc.fallbackUpsert != nil {
-			mockFallback.EXPECT().Upsert(gomock.Not(context.TODO()), adaptedEi, tc.fallbackUpsert.values).Return(nil)
+			// use gomock.Any() because gob encoding of maps is non-deterministic so could be a different argument value every time
+			mockFallback.EXPECT().Upsert(gomock.Not(context.TODO()), adaptedEi, gomock.Any()).Return(nil).AnyTimes()
 		}
 
-		connector := NewConnector(mockOrigin, mockFallback, tc.encoder, nil, cacheableEntities...)
+		connector := NewConnector(mockOrigin, mockFallback, nil, cacheableEntities...)
 		connector.setSynchronousMode(true)
 		err := connector.Upsert(context.TODO(), testEi, tc.originUpsert.values)
 		assert.NoError(t, err, tc.description)
 	}
 
+	var testBool bool
+
 	testCases := []testCase{
 		{
 			description: "Successful origin upsert also upserts to fallback",
-			encoder:     encoding.NewJSONEncoder(),
 			originUpsert: &expectArgs{
 				values: map[string]dosa.FieldValue{
 					"an_uuid_key": "d1449c93-25b8-4032-920b-60471d91acc9",
 					"strkey":      "test key string",
 					"StrV":        "test value string",
-					"BoolV":       false,
+					"BoolVP":      &testBool,
 				}},
 			fallbackUpsert: &expectArgs{
 				values: map[string]dosa.FieldValue{
-					key:   []byte(`[{"an_uuid_key":"d1449c93-25b8-4032-920b-60471d91acc9"},{"strkey":"test key string"}]`),
-					value: []byte(`{"BoolV":false,"StrV":"test value string","an_uuid_key":"d1449c93-25b8-4032-920b-60471d91acc9","strkey":"test key string"}`),
+					// gob encoding of [{"an_uuid_key":"d1449c93-25b8-4032-920b-60471d91acc9"},{"strkey":"test key string"}]
+					key: []byte{13, 255, 131, 2, 1, 2, 255, 132, 0, 1, 255, 130, 0, 0, 14, 255, 129, 4, 1, 2, 255, 130, 0, 1, 12, 1, 16, 0, 0, 98, 255, 132, 0, 2, 1, 11, 97, 110, 95, 117, 117, 105, 100, 95, 107, 101, 121, 6, 115, 116, 114, 105, 110, 103, 12, 38, 0, 36, 100, 49, 52, 52, 57, 99, 57, 51, 45, 50, 53, 98, 56, 45, 52, 48, 51, 50, 45, 57, 50, 48, 98, 45, 54, 48, 52, 55, 49, 100, 57, 49, 97, 99, 99, 57, 1, 6, 115, 116, 114, 107, 101, 121, 6, 115, 116, 114, 105, 110, 103, 12, 17, 0, 15, 116, 101, 115, 116, 32, 107, 101, 121, 32, 115, 116, 114, 105, 110, 103},
```

Comment:
Please change this test. You should just encode, and then decode, and compare. Please do not rely on the exact byte output, as this can change due to a bug fix in the gob encoder.

We don't care what the actual bytes contain, we just want to be sure the encoder and decoder work correctly.

Another approach: this test can be redone if you allow for the encoder to be changed. The encoder can just produce a mock value, and you can check that it returns that.